File: gui/src/app/core/models/AuditLog.ts
import {AuditLogType} from './enums/AuditLogType';

export class AuditLog {

  constructor(item: any) {
    this.id = item.id;
    this.requestId = item.requestId;
    this.actorId = item.actorId;
    this.actorName = item.actorName;
    this.type = AuditLogType[item.type as keyof typeof AuditLogType];
    this.madeAt = item.madeAt;
  }

  id: number = null;
  requestId: number = null;
  actorId: number = null;
  actorName: string = '';
  type: AuditLogType = AuditLogType.UNKNOWN;
  madeAt: string;

}

File: src/main/java/cz/metacentrum/perun/spRegistration/rest/models/AuditLog.java
package cz.metacentrum.perun.spRegistration.rest.models;

import cz.metacentrum.perun.spRegistration.common.enums.AuditMessageType;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NonNull;
import lombok.Setter;
import lombok.ToString;

import java.sql.Timestamp;

@Getter
@Setter
@ToString
@EqualsAndHashCode(exclude = {"madeAt"})
public class AuditLog {

    @NonNull private Long requestId;
    @NonNull private Long actorId;
    @NonNull private String actorName;
    @NonNull private AuditMessageType type;
    @NonNull private Timestamp madeAt;

}
